modules/020-deckhouse/hooks/update_deckhouse_image.go

```diff
@@ -321,9 +419,9 @@ func releaseChannelUpdate(input *go_hook.HookInput, releases []deckhouseReleaseU
 			}
 
 			// always deploy Patch releases
-			if !rl.StatusApproved && !isPatch {
+			if !rl.StatusApproved && !du.hasPatchRelease {
 				input.LogEntry.Infof("Release %s is waiting for manual approval", rl.Name)
-				input.MetricsCollector.Set("d8_release_waiting_manual", 1, map[string]string{"name": rl.Name}, metrics.WithGroup("d8_releases"))
+				input.MetricsCollector.Set("d8_release_waiting_manual", float64(du.totalPendingManualReleases), map[string]string{"name": rl.Name}, metrics.WithGroup("d8_releases"))
```

Comment:
So if I get it right, we expose the metric for all names with the same value?
```
d8_release_waiting_manual{name="v1-28-0"} 3
d8_release_waiting_manual{name="v1-29-0"} 3
d8_release_waiting_manual{name="v1-30-0"} 3
```

I am puzzled a little because it does not seem obvious what exactly these values represent.
Maybe if we look into the code, we will be able to get the meaning. However, it is impossible if we only see these metrics on Prometheus.

Reply:
this metric will be created only for the first pending release
```
d8_release_waiting_manual{name="v1-28-0"} 3
```

after this release is approved, the metric will be deleted
```
Expire("d8_releases")
```
and we will have a new metric

```
d8_release_waiting_manual{name="v1-29-0"} 2
```

That has been done because we have to apply releases in order. And we have instructions in this alert only for the next release. So we export the total count of manual pending releases but put labels only for the first one

Reply:
Also, we have a `return` after this exporting metric. So we place it only one time